src/app/recover-funds/[address]/RecoverFunds.tsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+import { Address } from '@arbitrum/sdk';
+import { BigNumber, utils, constants } from 'ethers';
+import { useEffect, useState } from 'react';
+import { useNetwork } from 'wagmi';
+import {
+  getProviderFromChainId,
+  getTargetChainId,
+  supportedL1Networks,
+} from '../../../lib';
+import { RecoverFundsButton } from './RecoverFundsButton';
+
+interface OperationInfo {
+  balanceToRecover: BigNumber;
+  aliasedAddress: string;
+}
+
+async function getData(
+  chainID: number,
+  address: string,
+): Promise<OperationInfo | null> {
+  // First, obtain the aliased address of the signer
+  const signerAddress = new Address(address);
+  const { value: aliasedAddress } = signerAddress.applyAlias();
+
+  // And get its balance to find out the amount we are transferring
+  try {
+    const l2Provider = getProviderFromChainId(chainID);
+    const aliasedSignerBalance = await l2Provider.getBalance(aliasedAddress);
+
+    return {
+      balanceToRecover: aliasedSignerBalance.eq(0)
+        ? constants.Zero
+        : aliasedSignerBalance,
+      aliasedAddress,
+    };
+  } catch (e) {
+    return {
+      balanceToRecover: constants.Zero,
+      aliasedAddress,
+    };
+  }
+}
+
+const RecoverFunds = ({ address }: { address: string }) => {
+  const { chain } = useNetwork();
+  const [operationInfo, setOperationInfo] = useState<OperationInfo | null>(
+    null,
+  );
+  const targetChainID = getTargetChainId(chain?.id);
+
+  useEffect(() => {
+    if (!targetChainID) {
+      return;
+    }
+
+    getData(targetChainID, address).then((data) => {
+      setOperationInfo(data);
+    });
+  }, [address, targetChainID]);
+
+  // No funds to recover
+  if (true && chain && operationInfo?.balanceToRecover.eq(0)) {
+    return (
+      <div className="funds-message">
+        There are no funds stuck on {operationInfo.aliasedAddress} (Alias of{' '}
+        {address}) on this network ({targetChainID}).
+      </div>
+    );
+  }
+
+  // Funds found on aliased address
+  if (chain && operationInfo?.balanceToRecover.gt(0)) {
```

Comment:
There will always be some leftover in the aliased account, because we can't predict exactly how much gas we are going to spend in the tx.
So this will always return true from the first time an account has funds in its aliased counterpart.
One possible solution is to set a minimum threshold, so we initiate the recover flow if the balanceToRecover is higher than that threshold.

(Same with the condition above this one)